src/pages/instances/forms/UploadExternalFormatFileForm.tsx

```diff
@@ -0,0 +1,335 @@
+        <Input
+          id="image-file"
+          name="imageFile"
+          type="file"
+          label="External format (.qcow2, .vmdk, etc...)"
+          accept=".img, .qcow, .qcow2, .vdi, .vhdx, .vmdk"
+          labelClassName="u-hide"
+          onChange={(e) => void handleFileChange(e)}
+        />
+        <Input
+          {...formik.getFieldProps("name")}
+          id="name"
+          type="text"
+          label="New instance name"
+          placeholder="Enter name"
+          error={formik.touched.name ? formik.errors.name : null}
+          disabled={!!noFileSelectedMessage}
+          title={noFileSelectedMessage}
+        />
+        <StoragePoolSelector
+          project={project?.name || ""}
+          value={formik.values.pool}
+          setValue={(value) => void formik.setFieldValue("pool", value)}
+          selectProps={{
+            id: "pool",
+            label: "Root storage pool",
+            disabled: !project || !!noFileSelectedMessage,
+            title: noFileSelectedMessage,
+          }}
+        />
+        <Select
+          {...formik.getFieldProps("architecture")}
+          id="architecture"
+          label="Image architecture"
+          options={archOptions}
+          disabled={!!noFileSelectedMessage}
+          title={noFileSelectedMessage}
+        />
+        <label htmlFor="">Conversion options</label>
+        <Input
+          {...formik.getFieldProps("formatConversion")}
+          type="checkbox"
+          label={
+            <span title={noFileSelectedMessage}>
+              Convert to raw format{" "}
+              <Icon
+                name="information"
+                title={
+                  noFileSelectedMessage
+                    ? noFileSelectedMessage
+                    : "Can be skipped if the image is already in raw format to speed up the import."
+                }
+              />
+            </span>
+          }
+          disabled={!!noFileSelectedMessage}
+          checked={formik.values.formatConversion}
+        />
+        <Input
+          {...formik.getFieldProps("virtioConversion")}
+          type="checkbox"
+          label={
+            <span title={noFileSelectedMessage}>
+              Add Virtio drivers{" "}
+              <Icon
+                name="information"
+                title={
+                  noFileSelectedMessage
+                    ? noFileSelectedMessage
+                    : "Mandatory, if the image does not have Virtio drivers installed."
+                }
+              />
+            </span>
+          }
+          disabled={!!noFileSelectedMessage}
+          checked={formik.values.virtioConversion}
+        />
+      </Form>
+      <footer className="p-modal__footer" id="modal-footer">
+        <Button
+          appearance="base"
+          className="u-no-margin--bottom"
+          type="button"
+          onClick={handleCloseModal}
+        >
+          Cancel
+        </Button>
```

Comment:
We should also cancel the upload through websocket process on cancel. Currently after "cancel", the upload continues in the background and I think the success callback gets triggered.

Reply:
@edlerd this is done, works nicely I think, thanks for the suggestion for removing the operation id, didn't even know that function existed